extensions/s3/src/main/java/io/deephaven/extensions/s3/AwsCredentialsImpl.java

```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.extensions.s3;
+
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+
+interface AwsCredentialsImpl extends AwsCredentials {
```

Comment:
Why is this a separate interface from `AwsCredentials`? Feels awfully weird to have an interface named `Impl`.

Reply:
It's the pattern I suggested to Shivam to keep aws library as impl instead of api. This could be a package private abstract base class, but interface is just as good or better imo bc you can use it w package private enum.